sbi/inference/trainers/vfpe/base_vf_inference.py

```diff
@@ -646,3 +489,181 @@ def _loss(
 
         assert_all_finite(loss, f"{cls_name} loss")
         return calibration_kernel(x) * loss
+
+    def _get_training_losses(self, batch: Any, loss_kwargs: Dict[str, Any]) -> Tensor:
+        """
+        Compute training losses for a batch of data.
+
+        Args:
+            batch: A batch of data.
+            loss_kwargs: Additional keyword arguments passed to self._loss fn.
+
+        Returns:
+            A tensor containing the computed training losses for each sample
+            in the batch.
+        """
+
+        # Skip validation_times as it is only used in _get_validation_losses method
+        loss_kwargs = {k: v for k, v in loss_kwargs.items() if k != "validation_times"}
+
+        # Get batches on current device.
+        theta_batch, x_batch, masks_batch = (
+            batch[0].to(self._device),
+            batch[1].to(self._device),
+            batch[2].to(self._device),
+        )
+
+        train_losses = self._loss(
+            theta=theta_batch,
+            x=x_batch,
+            masks=masks_batch,
+            **loss_kwargs,
+        )
+        return train_losses
+
+    def _get_validation_losses(self, batch: Any, loss_kwargs: Dict[str, Any]) -> Tensor:
```

Comment:
I see, there the two functions have a bit different logic, but maybe this could be handled via the loss kwargs? not sure... 
As always, this is a question of how much complexity the reduction of code duplication adds. if it gets complicated it's probably better to just have these two different functions.